src/platform/backends/lxd/lxd_vm_image_vault.cpp

```diff
@@ -275,6 +281,13 @@ bool mp::LXDVMImageVault::has_record_for(const std::string& name)
     {
         return false;
     }
+    catch (const LocalSocketConnectionException& e)
+    {
+        mpl::log(mpl::Level::warning, category,
+                 fmt::format("{} - Unable to determine if \'{}\' exists", e.what(), name));
+        // Assume instance exists until it knows for sure
+        return true;
```

Comment:
This looks weird, shouldn't we instead raise the error further to avoid what `return false` would result in?

Reply:
I considered that, but then it will require quite a bit of change in the `Daemon` class' ctor and felt it was way too much effort to basically accomplish what this does.  In the end, we know there is supposed to be an instance, but we cannot determine it's state, so we set it `UNKNOWN` and it's not startable.

Reply:
Right, sounds like a bigger refactoring with a common `mp::ShrugException`. Shall we have a TODO or an issue?

Reply:
I honestly don't think it's necessary to refactor as in the end, it will accomplish exactly the same thing as far as the daemon is concerned.  All we are avoiding here is for the daemon to remove the instance from it's internal map along with the removing it from the JSON.  We don't want it to do that until it knows for sure what the status of the instance is from LXD.

Reply:
The method name is `has_record_for()`. Returning `true` when we don't actually know is just not something I'm comfortable with in the long run :)

Renaming the function to `should_keep_instance()`, I could maybe be convinced with, but it feels like moving too much of the logic into the vault.

I'd be happiest with introducing an exception type common to all vaults, converting the error here to that one, or a child exception type, and interpreting that in the daemon. That doesn't need to happen in this PR, though.

Reply:
Thinking about it, perhaps we really don't need this function at all for either vault.  I think that is what should be investigated first in the future.  Not sure a TODO is the right thing for this since it affects multiple places in the code.  I guess a bug would be better even though it's quite technical in nature.

Reply:
#1766 then.